app/backend/src/couchers/servicers/donations.py

```diff
@@ -0,0 +1,167 @@
+                item = {
+                    "price": config.get("STRIPE_RECURRING_PRODUCT_ID"),
+                    "quantity": request.amount,
+                }
+            else:
+                item = {
+                    "price_data": {
+                        "currency": "usd",
+                        "unit_amount": request.amount * 100,  # input is in cents
+                        "product_data": {
+                            "name": f"Couchers financial supporter (one-time)",
+                            "images": ["https://couchers.org/img/share.jpg"],
+                        },
+                    },
+                    "quantity": 1,
+                }
+
+            checkout_session = stripe.checkout.Session.create(
+                client_reference_id=user.id,
+                submit_type="donate" if not request.recurring else None,
+                customer=user.stripe_customer_id,
+                success_url=urls.donation_success_url(),
+                cancel_url=urls.donation_cancelled_url(),
+                payment_method_types=["card"],
+                mode="subscription" if request.recurring else "payment",
+                line_items=[item],
+            )
+
+            if request.recurring:
+                session.add(
+                    RecurringDonation(
+                        user_id=user.id,
+                        amount=request.amount,
+                        stripe_checkout_session_id=checkout_session.id,
+                    )
+                )
+            else:
+                session.add(
+                    OneTimeDonation(
+                        user_id=user.id,
+                        amount=request.amount,
+                        stripe_checkout_session_id=checkout_session.id,
+                        stripe_payment_intent_id=checkout_session.payment_intent,
+                        paid=None,
+                    )
+                )
+
+            return donations_pb2.InitiateDonationRes(stripe_checkout_session_id=checkout_session.id)
+
+
+class Stripe(stripe_pb2_grpc.StripeServicer):
+    def Webhook(self, request, context):
+        # We're set up to receive the following webhook events (with explanations from stripe docs):
+        # For both recurring and one-off donations, we get a `checkout.session.completed` event, and then a `payment_intent.succeeded` event
+        headers = dict(context.invocation_metadata())
+
+        try:
+            logger.info(request.data)
+            event = stripe.Webhook.construct_event(
+                payload=request.data,
+                sig_header=headers.get("stripe-signature"),
+                secret=config.get("STRIPE_WEBHOOK_SECRET"),
+            )
+            data = event["data"]
+            event_type = event["type"]
+            event_id = event["id"]
+            data_object = data["object"]
+        except Exception as e:
+            context.abort(grpc.StatusCode.FAILED_PRECONDITION, errors.STRIPE_WEBHOOK_FAILED_SIG)
```

Comment:
Specify which exceptions that we could expect (I guess `KeyError`)

Reply:
Do we even really need to catch it here? We already have functionality for catching general exceptions and they'll end up as useful logs in our tracing?

Reply:
No, I guess stripe doesn't care about our HTTP return code value anyway. Just remove all error handing.